packages/babel-traverse/src/scope/index.js

```diff
@@ -740,6 +743,10 @@ export default class Scope {
     path.traverse(collectorVisitor, state);
     this.crawling = false;
 
+    if (state.unnamedDefault) {
+      state.unnamedDefault.id = this.generateUidIdentifier("default");
```

Comment:
> Those are actually changing the logic of the code.

And this changes any logic relying on `fn.name` from `default` to `_default`. This could just as well change the logic of something. It likely won't _now_ because nothing has ES6-native modules yet, but that doesn't mean things won't in the future.

> This is normalizing an AST into a coherent interface

It's normalizing it into a definition of coherent you've defined. I'd define it differently. The spec does not require names on certain declarations, therefore code working with declarations should be responsible for handling that.

> there's no where else to do it and every transform now has to worry about a stupid edge case

Every transform _should_ have to handle it, just like they have to handle any other valid state in the AST. 

If you wrote an transform and assumed all function params were `Identifier` nodes and now they can be default-assignments and spread params, stuff would break. We added those in, and people know to expect them now. People will learn to expect that not all function declarations have names, or they will get bug reports and fix them.

I don't see fear of bugs as a reason to mutate the AST at unexpected times. If bugs are the concern, let's add clearer error messaging around declaration naming, so errors are thrown earlier with a clear reason if a name isn't present when it should be. If bugs come up, plugin authors can fix them, it's really not that bad.